pointmatcher/DataPointsFilters/utils/utils.h

```diff
@@ -41,6 +41,18 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 namespace utils {
 
+template<class T>
+inline constexpr T pow(const T base, const std::size_t exponent)
+{
+    // (parentheses not required in next line)
+    return (exponent == 0)     ? 1 :
+           (exponent % 2 == 0) ? pow(base, exponent/2)*pow(base, exponent/2) :
+           base * pow(base, (exponent-1)/2) * pow(base, (exponent-1)/2);
+}
+template < typename T, T base, std::size_t exponent >
+using pow_ = std::integral_constant<T, pow(base, exponent)>;
```

Comment:
I don't see the need for this alias template. Why not use the pow function itself instead?

Reply:
I took the code from [here](https://stackoverflow.com/a/16443849).

The author suggest that 
>the compiler isn't required to do constant-initialization at compile-time, but if you use the result of a constexpr function as a non-type template argument, it has to compute the result at compile-time.

I should cite the link in a comment above the function, I forgot it !